### PR TITLE
chore: remove function total_n

### PR DESCRIPTION
**Pull request by mandytrinh**
**Summary:**
The function `total_n` yields a `TypeError` since it tries to subtract two lists

Example:
```
>>> posterior = {
... "log_marginal_likelihood_M1": -23.585991739528254,
... "log_marginal_likelihood_M0": -76546.65811250894,
... "posterior_M1": [88519.5, 12540.25, 13002.25],
... "posterior_M0": [0.75, 0.125, 0.125],
... }
>>> np.sum(posterior["posterior_M1"] - posterior["posterior_M0"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'list' and 'list'
```

We can fix this by casting the lists to `numpy` array first, e.g:
`>>> np.sum(np.array(posterior["posterior_M1"]) - np.array(posterior["posterior_M0"]))`

However, Michael recommends to remove this function altogether, since it is not used anywhere, so this PR does that.

### CODE REVIEW

File: ssrm_test/test_ssrm_test.py [hunk 1]
```diff
@@ -80,6 +81,17 @@ def test_overflow():
...
 
 
+def test_total_n():
+    posterior = {
+        "log_marginal_likelihood_M1": -23.585991739528254,
+        "log_marginal_likelihood_M0": -76546.65811250894,
+        "posterior_M1": [88519.5, 12540.25, 13002.25],
+        "posterior_M0": [0.75, 0.125, 0.125],
+    }
+    actual = total_n(posterior)
```
Comment (joetang631) [inline ssrm_test/test_ssrm_test.py:91]: Just curious that it seems like the total_n method only parse two dict keys, which is posterior_M1 and posterior_M0. Any reason the posterior include log_marginal_likelihood_M1 and log_marginal_likelihood_M0.?

File: ssrm_test/ssrm_test.py [hunk 2]
```diff
@@ -159,7 +159,9 @@ def total_n(posterior: dict) -> float:
     float
         Total number of datapoints in the posterior.
     """
-    return np.sum(posterior["posterior_M1"] - posterior["posterior_M0"])
+    posterior_M1 = np.array(posterior["posterior_M1"])
+    posterior_M0 = np.array(posterior["posterior_M0"])
+    return np.sum(posterior_M1 - posterior_M0)
```
Comment (michaellindon) [inline ssrm_test/ssrm_test.py:164]: I think we should Just delete the total_n function, it's no longer valid. If we look at the sequential_posteriors function, we can see that posterior_M1 is initialized as follows

```
  if dirichlet_probability is None:
        dirichlet_probability = null_probabilities
    dirichlet_alpha = dirichlet_probability * dirichlet_concentration
    acc = {
        "log_bayes_factor": 0,
        "bayes_factor": 1,
        "p_value": 1,
        "posterior_probability": 0.5,
        "log_marginal_likelihood_M1": 0,
        "log_marginal_likelihood_M0": 0,
        "posterior_M1": dirichlet_alpha,
        "posterior_M0": null_probabilities,
    }
```

the default value of dirichlet_concentration is 10000, so I'd expect total_n as it is currently coded to be off by 10000. The function isn't being used anyway so I would say delete it.

File: ssrm_test/test_ssrm_test.py (hunk 1 above)
Comment (mandytrinh) [reply ssrm_test/test_ssrm_test.py:91]: the param the function expects is `posterior` which is a dictionary with these 4 keys and values

File: ssrm_test/ssrm_test.py (hunk 2 above)
Comment (mandytrinh) [reply ssrm_test/ssrm_test.py:164]: I see, ok I will delete it.

**Pull request merged**